packages/components/src/buttons/src/ClearButton.tsx

```diff
@@ -0,0 +1,85 @@
+import { DismissIcon } from "@hopper-ui/icons";
+import {
+    type StyledComponentProps,
+    useStyledSystem
+} from "@hopper-ui/styled-system";
+import { type ForwardedRef, forwardRef } from "react";
+import {
+    Button as RACButton,
+    type ButtonProps as RACButtonProps,
+    composeRenderProps,
+    useContextProps
+} from "react-aria-components";
+
+import { useLocalizedString } from "../../i18n/index.ts";
+import {
+    composeClassnameRenderProps,
+    cssModule
+} from "../../utils/index.ts";
+
+import { ClearButtonContext } from "./ClearButtonContext.ts";
+
+import styles from "./ClearButton.module.css";
+
+export const GlobalClearButtonCssSelector = "hop-ClearButton";
+
+export interface ClearButtonProps extends StyledComponentProps<Omit<RACButtonProps, "children">> {}
+
+function ClearButton(props: ClearButtonProps, ref: ForwardedRef<HTMLButtonElement>) {
+    [props, ref] = useContextProps(props, ref, ClearButtonContext);
+
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const stringFormatter = useLocalizedString();
+
+    const {
+        "aria-label": ariaLabel = stringFormatter.format("ClearButton.clearAriaLabel"),
+        className,
+        isDisabled,
+        style: styleProp,
+        slot = "remove",
```

Comment:
hmm assigning a slot after useContextProps is a bit odd. 

Since the only purpose of the slot at the moment is assigning it to the dom, but it has no use. 

You should remove it from this component, and pass it  as a prop when needed